android/src/main/java/io/mosip/tuvali/common/events/EventProducer.kt

```diff
@@ -0,0 +1,24 @@
+package io.mosip.tuvali.common.events
+
+import io.mosip.tuvali.common.events.withArgs.ErrorEvent
+import io.mosip.tuvali.exception.ErrorCode
+
+class EventProducer {
+  private var consumers = mutableListOf<((Event) -> Unit)>()
+
+  fun addConsumer(consumer: (Event) -> Unit) {
+    consumers.add(consumer)
+  }
+
+  fun emitErrorEvent(message: String, code: ErrorCode) {
+    emitEvent(ErrorEvent(message, code.value))
+  }
+
+  fun emitEvent(event: Event) {
+    consumers.forEach{it(event)}
+  }
+
+  fun removeConsumer() {
```

Comment:
make it plural `removeConsumers` or `clearConsumers`

Reply:
done